File: problem3.js
var test = 900000000001 ;

var primeFactors = function(number) {
    factors = [];
	divisor = 2;
	while ( number > 1 ) {
		while ( number % divisor === 0 ){
			factors.push(divisor);
			number = number / divisor ;
		}

		divisor++ ;
	}

	return factors;
};

pFArray = primeFactors(test);
var answer = Math.max.apply(Math, pFArray);
console.log(answer);

File: problem3.py
input = raw_input("Please enter a number: ")
test = float(input)

def primeFactors(number):
	
	factors = []
	divisor = 2
	
	while number > 1:
		while number % divisor == 0:
			factors.append(divisor)
			number = number / divisor
		
		divisor = divisor + 1
	
	return factors

pFArray = primeFactors(test)
answer = max(pFArray)
print answer
